test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -978,34 +978,38 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
 
     void test_tag_send_recv(size_t size, bool is_exp, bool is_sync = false)
     {
-        std::string send_buf(size, 'x');
-        std::string recv_buf(size, 'y');
+        /* send multiple messages to test the protocol both before and after
+         * connection establishment */
+        for (int i = 0; i < 100; i++) {
```

Comment:
@alinask @yosefe do we want the same logic for RMA and AM (to test RNDV)?

Reply:
I have a dependency on this change. @alinask what about creating a separate PR with sockaddr-only change to test before/after connection establishment? If you wish, I could help with that

Reply:
@dmitrygx which PR is dependent on this?
if you add only this change, of a loop around this code, it will fail on the ptr_map leak which this PR is fixing (and this change is reproducing).

Reply:
> @dmitrygx which PR is dependent on this?
> if you add only this change, of a loop around this code, it will fail on the ptr_map leak which this PR is fixing (and this change is reproducing).

I see
yes, #5821 depends on the tests, but with error handling